urlparse4/cgurl.pyx

```diff
@@ -333,13 +392,157 @@ def urljoin(base, url, allow_fragments=True):
     This function intends to replace urljoin from urllib,
     which uses Resolve function from class GURL of GURL chromium
     """
+    str_input = isinstance(base, str)
+    if isinstance(url, str) != str_input:
+        raise TypeError("Cannot mix str and non-str arguments")
+
     decode = not (isinstance(base, bytes) and isinstance(url, bytes))
     if allow_fragments and base:
         base, url = unicode_handling(base), unicode_handling(url)
+        """
+        this part needs to be profiled to see if creating another GURL instance
+        here takes more time than expected?
```

Comment:
yeah, I would definitely investigate, most likely it's better to remove double-parsing